lib/proc/proc_snippet.c

```diff
@@ -251,19 +261,55 @@ func_snippet_html(grn_ctx *ctx, int nargs, grn_obj **args,
   grn_obj *snippets = NULL;
 
   /* TODO: support parameters */
-  if (nargs == 1) {
+  if (nargs > 0) {
     grn_obj *text = args[0];
     grn_obj *expression = NULL;
     grn_obj *condition_ptr = NULL;
     grn_obj *condition = NULL;
     grn_obj *snip = NULL;
+    grn_obj *default_return_value = NULL;
     int flags = GRN_SNIP_SKIP_LEADING_SPACES;
     unsigned int width = 200;
     unsigned int max_n_results = 3;
     const char *open_tag = "<span class=\"keyword\">";
     const char *close_tag = "</span>";
     grn_snip_mapping *mapping = GRN_SNIP_MAPPING_HTML_ESCAPE;
 
+    int i;
+    for (i = 1; i < nargs; i++) {
```

Comment:
Why should we need to loop here?

Reply:
Ah...This loop is needless. Therefore I've removed it.